File: backend/src/main/java/com/pickleball/backend/modules/court/service/CourtService.java
package com.pickleball.backend.modules.court.service;

import com.pickleball.backend.modules.court.dto.request.CreateCourtRequest;
import com.pickleball.backend.modules.court.dto.request.UpdateCourtRequest;
import com.pickleball.backend.modules.court.dto.response.CourtResponse;
import com.pickleball.backend.response.PageResponse;

public interface CourtService {

    PageResponse<CourtResponse> getCourts(
            Long clubId, String search, 
            java.math.BigDecimal minPrice, java.math.BigDecimal maxPrice, 
            java.time.LocalDate date, java.time.LocalTime startTime, java.time.LocalTime endTime,
            Double userLat, Double userLng, Double radiusInKm,
            String province, String district,
            int page, Integer size, String sort);

    CourtResponse getCourtById(Long id);

    CourtResponse createCourt(String userEmail, CreateCourtRequest request);

    CourtResponse updateCourt(String userEmail, Long id, UpdateCourtRequest request);

    void deleteCourt(Long id);
}

File: frontend/src/services/court.service.ts
import { apiClient } from "@/lib/axios";
import { Court, PageResponse } from "@/types/court.type";

export const courtService = {
  getCourts: async (params?: { search?: string; clubId?: number; page?: number; size?: number }) => {
    const res = await apiClient.get<{ data: PageResponse<Court> }>("/courts", {
      params,
    }) as unknown as { data: PageResponse<Court> };
    return res.data;
  },

  getCourtById: async (id: number) => {
    const res = await apiClient.get<{ data: Court }>(`/courts/${id}`) as unknown as { data: Court };
    return res.data;
  },

  createCourt: async (data: Partial<Court>) => {
    const res = await apiClient.post<{ data: Court }>("/courts", data) as unknown as { data: Court };
    return res.data;
  },

  updateCourt: async (id: number, data: Partial<Court>) => {
    const res = await apiClient.put<{ data: Court }>(`/courts/${id}`, data) as unknown as { data: Court };
    return res.data;
  },

  deleteCourt: async (id: number) => {
    await apiClient.delete(`/courts/${id}`);
  },
};
